### PR TITLE
feat(remote-host): list accounting books + choose role on startChat

### PR DESCRIPTION
**Pull request by snakajima**
## What

Two new capabilities on the mobile-remote command channel (`server/remoteHost/handlers/`), the RPC-over-Firestore layer the phone reaches via `callHost(...)`.

### 1. List accounting books (id + name)
- New handler `listAccountingBooks` → returns `{ books: [{ id, name }] }` so the remote can render a book picker.
- Runs in-process on the host, bypassing the HTTP bearer layer; only `id` + `name` travel (drops currency/country/fiscalYearEnd/createdAt) to keep the command payload minimal — same discipline as `listSkills`.
- Exports `listBooks` (+ `BookSummary` type) from `@mulmoclaude/accounting-plugin/server` so the host can enumerate books outside the HTTP dispatch route.
- Registered in the handler table.

### 2. Specify the role when starting a chat
- `startChat` now accepts an optional `role` param (a role id). Absent / `null` / `""` → the default `general` role (byte-for-byte the prior behaviour).
- A provided id is validated against `loadAllRoles()` (built-in **and** custom roles, so `accounting` works); an unknown or non-string role rejects the command **before** any ingest/spawn.
- This intentionally reverses the handler's old "role is NOT part of the remote API" stance — the header comment is updated to document the new contract.

## Tests
Unit tests added: the `{ id, name }` mapping (+ empty list) and the default / known / unknown / non-string role branches. Full suite green (`yarn test`), plus `format` / `lint` / `typecheck` / `build`.

## Follow-up (separate repo)
The phone client lives in the sibling **`mulmoserver`** repo. To use these it needs matching calls: `callHost(..., "listAccountingBooks")` for the picker, and a `role` field added to its `startChat` params.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by Sourcery

Expose a new remote-host command for listing accounting books and allow specifying a validated role when starting a chat over the remote command channel.

New Features:
- Add a listAccountingBooks remote-host handler that returns a minimal list of accounting books for mobile clients.
- Export listBooks and the BookSummary type from the accounting plugin server API for host-side enumeration of books outside HTTP routes.
- Allow the startChat remote-host handler to accept an optional role parameter to run chats under a selected role.

Bug Fixes:
- Validate provided role IDs against the role registry in startChat and reject unknown or non-string roles before spawning chats.

Tests:
- Add unit tests for the listAccountingBooks handler mapping and empty results.
- Extend startChat tests to cover default role behaviour, valid custom roles, and invalid role inputs.

## Summary by CodeRabbit

* **New Features**
  * Added a new remote-host command to list available accounting books (for picker-style flows).
  * Chat creation now supports selecting a role when starting a conversation.
  * Exposed accounting book listing in the accounting plugin’s public API for non-HTTP integrations.
* **Bug Fixes**
  * Improved role handling: missing/empty roles fall back to the default; unknown or debug-only roles are rejected before chat starts.
* **Tests**
  * Added coverage for the new book-listing command and role-selection behavior.

### CODE REVIEW

File: server/remoteHost/handlers/startChat.ts [hunk 1]
```diff
@@ -101,6 +105,17 @@ const composeCollectionSeed = async (deps: StartChatDeps, params: JsonObject, me
...
 const hasSlug = (value: JsonValue | undefined): boolean => value != null && value !== "";
 
+// Resolve the optional `role` param to a concrete roleId. Absent / null / "" ⇒
+// the host default. A provided id must be a string that matches an existing
+// role (built-in or custom) — reject an unknown one so we never seed the chat
+// with the wrong (default-fallback) assistant.
+const resolveRoleId = (deps: StartChatDeps, role: JsonValue | undefined): string => {
+  if (role == null || role === "") return DEFAULT_ROLE_ID;
+  if (typeof role !== "string") throw new Error("role must be a string");
+  if (!deps.loadRoles().some((candidate) => candidate.id === role)) throw new Error(`role '${role}' not found`);
```
[the remark this answers is not in the document]
Comment (snakajima) [reply server/remoteHost/handlers/startChat.ts]: Good catch — fixed in 2f635265. `resolveRoleId` now excludes `isDebugRole` roles from the acceptable set, so a remote `{ role: "debug" }` is rejected as unknown. The remote channel is production-facing, so debug roles are gated unconditionally here (rather than mirroring the client-side `VITE_DEV_MODE` flag, which is a browser build-time env not available server-side). Added a regression test.

File: server/remoteHost/handlers/listAccountingBooks.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+export const createListAccountingBooks =
+  (deps: ListAccountingBooksDeps): CommandHandler =>
+  // Takes no params (the `__` prefix marks it intentionally unused per lint).
+  async (__params: JsonObject) => {
+    const { books } = await deps.listBooks();
+    // { id, name } are always present on a BookSummary. The cast only satisfies
+    // the channel's structural JsonValue type, which the BookSummary interface
+    // (no index signature) can't match directly.
+    return { books: books.map((book) => ({ id: book.id, name: book.name })) } as unknown as JsonObject;
```
[the remark this answers is not in the document]
Comment (snakajima) [reply server/remoteHost/handlers/listAccountingBooks.ts:28]: Keeping the `as unknown as JsonObject` cast for consistency with the sibling `listCollections.ts` handler, which uses the identical pattern and comment. `JsonObject` is an index-signature type (`{ [k: string]: JsonValue }`); a `{ id, name }` object literal is not structurally assignable to it, and the suggested `& JsonObject` intersection would still require a cast to construct. The cast is localized to the handler boundary and documented, so the trade-off is intentional.
